ts/features/newProfile/saga/profile.ts

```diff
@@ -0,0 +1,128 @@
+import { sessionTokenSelector } from "../../../store/reducers/authentication";
+import { authenticationSaga } from "../../../sagas/startup/authenticationSaga";
+import { refreshUserProfileDataRequest } from "../store/actions/profile";
+
+const BACKEND_PROFILE_RETRY_DELAY_INTERVAL = (6 * 1000) as Millisecond;
+const apiUrlPrefix: string = Config.API_URL_PREFIX;
+
+// This function listens for Profile refresh requests and calls the needed saga.
+export function* refreshProfileOnMountOfTheNewProfilePageSaga(): Iterator<ReduxSagaEffect> {
+  yield* takeLatest(getType(refreshUserProfileDataRequest), refreshProfile);
+}
+
+export function* refreshProfile(): Generator<ReduxSagaEffect, void, any> {
+  // Whether the user is currently logged in.
+  const previousSessionToken: ReturnType<typeof sessionTokenSelector> =
+    yield* select(sessionTokenSelector);
+
+  const sessionToken: SagaCallReturnType<typeof authenticationSaga> =
+    previousSessionToken
+      ? previousSessionToken
+      : yield* call(authenticationSaga);
+
+  const backendClient: ReturnType<typeof BackendClient> = BackendClient(
+    apiUrlPrefix,
+    sessionToken
+  );
+
+  // Load the profile info
+  const maybeUserProfile: SagaCallReturnType<typeof loadProfile> = yield* call(
+    loadProfile,
+    backendClient.getProfile
+  );
+
+  if (O.isNone(maybeUserProfile)) {
+    /**
+     * If we didn't succeed,
+     * we try again after waiting some time.
+     */
+    yield* delay(BACKEND_PROFILE_RETRY_DELAY_INTERVAL);
+    yield* put(refreshUserProfileDataRequest());
+  }
+}
+
+// A saga to load the Profile.
+export function* loadProfile(
+  getProfile: ReturnType<typeof BackendClient>["getProfile"]
+): Generator<
+  ReduxSagaEffect,
+  O.Option<InitializedProfile>,
+  SagaCallReturnType<typeof getProfile>
+> {
+  try {
+    const response = yield* call(getProfile, {});
+    const backendProfile = pipe(
+      response,
+      E.foldW(
+        reason => {
+          /**
+           * Here we comes even when we get a responso with a status error type,
+           * different from the ones of our return types.
+           * Here for example we manage only:
+           * - 200
+           * - 400
+           * - 401
+           * - 429
+           * - 500
+           */
+          throw Error(readablePrivacyReport(reason));
+        },
+        response => {
+          if (response.status === 200) {
+            return O.some(response.value);
+          }
+          if (response.status === 401) {
+            return O.none;
+          }
+          /**
+           * Here we come for all other "known" status code: 400, 500.
+           * 429 is known, but it seems to be managed differently (by the API client).
+           * For example, having a Server Error 500,
+           * but the JSON derserializer expect a JSON Body.
+           */
+          throw response
+            ? Error(`response status ${response.status}`)
+            : Error(I18n.t("profile.errors.load"));
```

Comment:
```suggestion
          throw Error(`response status ${response.status}`);
```

Just handling the generic error is enough since we don't need the localized string information in the redux-store

Reply:
Okay! 👍